twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,321 @@
+            cutRegion(C_minX,C_minY,R_minX,C_maxY,true,false).done(function(){
+            //X from right to left
+                   deferred.resolve();
+            }).fail(function(){
+
+                    clearCache().done();
+                    deferred.resolve();
+            })
+            return deferred.promise();
+
+       }else if(!LowerRight &&   !LowerLeft && !UpperLeft && !UpperRight){
+
+              cutRegion(C_minX,C_minY,C_maxX,C_maxY,true,true).done(function(){
+              //NO Overlap
+                     deferred.resolve();
+              }).fail(function(){
+
+                      clearCache().done();
+                      deferred.resolve();
+              })
+              return deferred.promise();
+       }
+ }
+//sees whether cutting the region can satisfy the target ,If not sends a failed message to evict function
+//X = true Y = false
+//TopTo Bottom = true
+//BottomToTop = false
+//LeftToRight = true
+//RightToLeft = false
+ var cutRegion =  function findCornerofEviction(minX,minY,maxX,maxY,XorY,Direction){
+
+            var deferred = new $.Deferred();
+            var line;
+            if(XorY){
+                if(Direction){
+                    //Left to Right
+                    line = turf.lineString([[minX,maxY],[maxX,maxY]]);
+                }else{
+                    //Right to Left
+                    line = turf.lineString([[maxX,maxY],[minX,maxY]]);
+                }
+            }else{
+                if(Direction){
+                    //TopToBottom
+                    line = turf.lineString([[maxX,maxY],[maxX,minY]]);
+                }else{
+                    //BottomToTop
+                    line = turf.lineString([[maxX,minY],[maxX,maxY]]);
+                }
+            }
+
+            var distance = turf.lineDistance(line, 'miles');
+            var Move = distance/10;
+            var start = 0;
+            var stop= distance;
+            var removeItems;
+            var sliced,cutPoint,cutBbox,remove_search;
+
+            while(DeletedCount<DeleteTarget){
+
+                  sliced = turf.lineSliceAlong(line, start, Move, 'miles');
+                  if(XorY){
+                  cutPoint = sliced["geometry"]["coordinates"][1][0];
+                  }else{
+                  cutPoint = sliced["geometry"]["coordinates"][1][1];
+                  }
+
+                  cutBbox = [minX,minY,cutPoint,maxY];
+                  remove_search = {
+                                        minX: cutBbox[0],
+                                        minY: cutBbox[1],
+                                        maxX: cutBbox[2],
+                                        maxY: cutBbox[3]
+                                    }
+                  removeItems = cachedCityPolygonTree.search(remove_search);
+                  DeletedCount = removeItems.length;
+                  Move += Move;
+                  if(Move>stop)
+                  {break;}
```

Comment:
You are right ,But  We have libraries to search incrementally,it is implementation wise very hard and time consuming to add regions(merge is expensive),Thats why search for bigger region each time .This sounded much better when time needed to be low.